crate/server/src/core/operations/import.rs

```diff
@@ -774,3 +777,83 @@ pub(crate) fn upsert_links_in_attributes(attributes: &mut Attributes, links_to_a
         attributes
     );
 }
+
+pub(crate) async fn process_secret_data(
+    kms: &KMS,
+    request: Import,
+    owner: &str,
+    params: Option<Arc<dyn SessionParams>>,
+) -> Result<(String, Vec<AtomicOperation>), KmsError> {
+    // check if the object will be replaced if it already exists
+    let replace_existing = request.replace_existing.unwrap_or(false);
+
+    // Generate a new UID if none is provided.
+    let uid = match request.unique_identifier.to_string() {
+        uid if uid.is_empty() => Uuid::new_v4().to_string(),
+        uid => uid,
+    };
+
+    let mut object = request.object;
+    // Unwrap the Object if required.
+    if request.key_wrap_type == Some(KeyWrapType::NotWrapped) {
```

Comment:
I'm still bugging when seeing an object with type `NotWrapped` that needs to be unwrap.
But it is correct apparently, so ignore this